src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -132,6 +134,42 @@ trait IterableOps[+A] extends Any {
   }
 
   override def toString = s"$className(${mkString(", ")})"
+
+  def sameElements[B >: A](that: Iterable[B]): Boolean = {
+  val these = this.iterator()
+  val those = that.iterator()
+  while (these.hasNext && those.hasNext)
+    if (these.next() != those.next())
+      return false
+    !these.hasNext && !those.hasNext
+  }
+
+  override def equals(o: scala.Any): Boolean =
+    o match {
+      case iterable: Iterable[A] => sameElements(iterable)
+      case _ => false
+    }
+
+  override def hashCode(): Int =
+    IterableUtils.orderedHash(coll, IterableUtils.iterableSeed)
+
+}
+
+// Temporary: TODO move to MurmurHash3.scala
+object IterableUtils {
+
+  val iterableSeed: Int = "Iterable".##
+
+  final def orderedHash(xs: Iterable[_], seed: Int): Int = {
+    var n = 0
+    var h = seed
+    xs foreach { x =>
+      h = MurmurHash3.mix(h, x.##)
+      n += 1
+    }
```

Comment:
Unless you have benchmarking results indicating that `foreach` is at least as fast as `iterator`, I would manually iterate this one to avoid needing to wrap `n` and `h` to pass into the closure.